src/number_theory/legendre_symbol/quadratic_char.lean

```diff
@@ -263,14 +273,65 @@ lemma quadratic_char_dichotomy {a : F} (ha : a ≠ 0) :
   quadratic_char F a = 1 ∨ quadratic_char F a = -1 :=
 (sq_eq_one_iff (quadratic_char F a)).mp (quadratic_char_sq_one ha)
 
+/-- For `a : F`, `quadratic_char F a = -1 ↔ ¬ is_square a`. -/
+lemma quadratic_char_neg_one_iff_not_is_square {a : F} :
+  quadratic_char F a = -1 ↔ ¬ is_square a :=
+begin
+  by_cases ha : a = 0,
+  { simp only [ha, is_square_zero, quadratic_char_zero, zero_eq_neg, one_ne_zero, not_true], },
+  { have h : quadratic_char F a = -1 ↔ ¬ quadratic_char F a = 1,
+    { split,
+      { intro h,
+        rw h,
+        norm_num, },
+      { exact λ (h₂ : ¬ quadratic_char F a = 1),
+                (or_iff_right h₂).mp (quadratic_char_dichotomy ha), }, },
+    rw h,
+    exact not_iff_not.mpr (quadratic_char_one_iff_is_square ha), },
+end
+
 /-- If `F` has odd characteristic, then `quadratic_char F` takes the value `-1`. -/
 lemma quadratic_char_exists_neg_one (hF : ring_char F ≠ 2) : ∃ a, quadratic_char F a = -1 :=
+Exists.dcases_on (finite_field.exists_nonsquare hF)
+  (λ (b : F) (h₁ : ¬is_square b), ⟨b, (quadratic_char_neg_one_iff_not_is_square.mpr h₁)⟩)
+
+/-- The number of solutions to `x^2 = a` is determined by the quadratic character. -/
+lemma quadratic_char_number_of_sqrts (hF : ring_char F ≠ 2) (a : F) :
```

Comment:
Should this have `card_sqrt` in its name?

Reply:
Is there a naming convention on this kind of thing? `card_sqrt` is at least shorter than `number_of_sqrts`...

Reply:
I think `card_sqrt` is fine. Certainly having `card` in the name is a win, because it also occurs in the statement.

Reply:
OK, I'll change the name to `quadratic_char_card_sqrts`.